generative_data_prep/data_prep/pipeline.py

```diff
@@ -402,6 +402,9 @@ def pipeline_main(  # noqa: C901
         test_ratio,
     )
 
+    lines = open(input_file_path).readlines()
```

Comment:
This will cause OOM on larger datasets, and would cause this repo to break for those cases.

Reply:
**It should work to check the input jsonl** instead of all the individual split files. But I would make sure there are no corner cases where the input jsonl has enough lines, but one of the splits for some reason does not have enough lines in it.